main.go

```diff
@@ -221,10 +221,12 @@ func getPublicURL(u *url.URL) *url.URL {
 	}
 	for _, a := range addrs {
 		if ipnet, ok := a.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
-			if ipnet.IP.To4() != nil {
-				r, _ := url.Parse(fmt.Sprintf("tcp://%v:%v", ipnet.IP.String(), u.Port()))
-				return r
+			urlSrting := fmt.Sprintf("tcp://%v:%v", ipnet.IP.String(), u.Port())
```

Comment:
Should we fix the issue here https://github.com/networkservicemesh/sdk/blob/main/pkg/tools/listenonurl/listenonurl.go and re-use it?